app/services/RegexMatcher.scala

```diff
@@ -0,0 +1,39 @@
+package services
+
+import model.{RegexRule, RuleMatch}
+import utils.Matcher
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Future
+
+class RegexMatcher(category: String, rules: List[RegexRule]) extends Matcher {
+  def getId() = "regex-validator"
+
+  override def check(request: MatcherRequest): Future[List[RuleMatch]] = {
+    Future.successful(rules.flatMap { checkRule(request, _) })
+  }
+
+  override def getRules(): List[RegexRule] = rules
+
+  override def getCategory(): String = category
+
+  private def checkRule(request: MatcherRequest, rule: RegexRule): List[RuleMatch] = {
+    request.blocks.flatMap { block =>
+      val iterator = rule.regex.findAllMatchIn(block.text)
```

Comment:
I think you can import `scala.collection.JavaConverters._` and convert using `.asScala` to call `.map` rather than manually running the iterator in a while loop

Reply:
Oh, that's _exactly_ what I was looking for. Thanks!